src/share/vm/gc_interface/allocTracer.inline.hpp

```diff
@@ -60,7 +60,7 @@ inline void AllocTracer::send_opto_array_allocation_event(KlassHandle klass, oop
   EventOptoArrayObjectAllocation event;
   if (event.should_commit()) {
     event.set_objectClass(klass());
-    event.set_address((TYPE_ADDRESS)obj);
+    event.set_address(cast_from_oop<TYPE_ADDRESS>(obj));
```

Comment:
Added. Fastdebug version has `-DCHECK_UNHANDLED_OOPS` which is a lack in slowdebug and release version. An `oop` will be treated as an object instead of a pointer for debugging purposes as we've seen. (ASNB)